File: Leetcode/Two_Sum.py
class Solution:
    def twoSum(self, arr, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        r = {}
        r[arr[0]] = 0
        for i in range(1,len(arr)):
            if (target - arr[i]) in r:
                return [i,r[target - arr[i]]]
            else:
                r[arr[i]] = i

File: Leetcode/Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> arr;
        vector<int> result;
        
        for(int idx=0; idx<nums.size(); idx++)
        {
            arr.insert({nums[idx], idx});
        }

        for(int i=0; i<nums.size(); i++)
        {
            int k = target - nums[i];
            auto iter = arr.find(k);
            if(iter == arr.end() || iter->second == i) { continue; }
            else
            {
                result.push_back(i);
                result.push_back(iter->second);
                break;
            }
        }
            
        return result;
    }
};